aws/rust-runtime/aws-inlineable/src/presigning.rs

```diff
@@ -0,0 +1,232 @@
+        /// by adding the `expires_in` duration to this start time. If not specified, this will
+        /// default to the current time.
+        ///
+        /// Optional.
+        pub fn set_start_time(&mut self, start_time: Option<SystemTime>) {
+            self.start_time = start_time;
+        }
+
+        /// Sets how long the request should be valid after the `start_time` (which defaults
+        /// to the current time).
+        ///
+        /// Required.
+        pub fn expires_in(mut self, expires_in: Duration) -> Self {
+            self.set_expires_in(Some(expires_in));
+            self
+        }
+
+        /// Sets how long the request should be valid after the `start_time` (which defaults
+        /// to the current time).
+        ///
+        /// Required.
+        pub fn set_expires_in(&mut self, expires_in: Option<Duration>) {
+            self.expires_in = expires_in;
+        }
+
+        /// Builds the `PresigningConfig`. This will error if `expires_in` is not
+        /// given, or if it's longer than one week.
+        pub fn build(self) -> Result<PresigningConfig, Error> {
+            let expires_in = self.expires_in.ok_or(Error::ExpiresInRequired)?;
+            if expires_in > Duration::from_secs(604800) {
+                return Err(Error::ExpiresInDurationTooLong);
+            }
+            Ok(PresigningConfig {
+                start_time: self.start_time.unwrap_or_else(SystemTime::now),
+                expires_in,
+            })
+        }
+    }
+}
+
+/// Presigned request
+pub mod request {
+    use std::fmt::{Debug, Formatter};
+
+    /// Represents a presigned request. This only includes the HTTP request method, URI, and headers.
+    #[non_exhaustive]
+    pub struct PresignedRequest(http::Request<()>);
+
+    impl PresignedRequest {
+        pub(crate) fn new(inner: http::Request<()>) -> Self {
+            Self(inner)
+        }
+
+        /// Returns the HTTP request method.
+        pub fn method(&self) -> &http::Method {
+            self.0.method()
+        }
+
+        /// Returns the HTTP request URI.
+        pub fn uri(&self) -> &http::Uri {
+            self.0.uri()
+        }
+
+        /// Returns any HTTP headers that need to go along with the request, except for `Host`,
+        /// which should be sent based on the endpoint in the URI by the HTTP client rather than
+        /// added directly.
+        pub fn headers(&self) -> &http::HeaderMap<http::HeaderValue> {
+            self.0.headers()
+        }
+    }
+
+    impl Debug for PresignedRequest {
+        fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+            f.debug_struct("PresignedRequest")
+                .field("method", self.method())
+                .field("uri", self.uri())
+                .field("headers", self.headers())
+                .finish()
+        }
+    }
+}
+
+/// Tower middleware service for creating presigned requests
+pub mod service {
```

Comment:
does this mod need to be pub? IIRC, you're making the parent module pub, so this could be private right?